Camera Sliders and real aspect ratio of acis

ho11laqe opened the pull request:
Added camera sliders and fixed ratio of the axis.
I multiplied the z-axis ratio by 2 so that the terrain looks twice as steep as it is

jouvetg commented:
Thanks Ethan for the fix, unfortunatly I still have an error when running python ~/IGM/igm-public/utils/prepare-data-from-rgi.py --preprocess False --border 5 --RGI RGI60-11.02822 --observation True

   line 239, in read_glathida
    thkobs[tuple(zip(*thickness_gridded.index))] = thickness_gridded
IndexError: index 5034898 is out of bounds for axis 0 with size 86

This time I have mot merged

ezwelty commented:
Sorry, my code contributions are usually of higher quality! But I haven't been able to run the script (errors with missing dependencies, OGGM internals, etc), so I had to guess the structure of the function arguments:

```py
import numpy as np

nrows, ncols = 45, 90
x = np.arange(ncols)
y = np.arange(nrows)
usurf = np.zeros((nrows, ncols))
proj = 'epsg:4326'
read_glathida(x, y, usurf, proj)
```

Did I make a wrong assumption?
The code above works. Here is a stand-alone version of the function for testing:

```py
import glob
import os

import numpy as np
from pyproj import Transformer


def read_glathida(x, y, usurf, proj):

    print("read_glathida ----------------------------------")

    from scipy.interpolate import RectBivariateSpline
    import pandas as pd

    if not os.path.exists("glathida"):
        os.system("git clone https://gitlab.com/wgms/glathida")

    files = ["glathida/data/point.csv"]
    files += glob.glob("glathida/submissions/*/point.csv")

    transformer = Transformer.from_crs(proj, "epsg:4326", always_xy=True)

    lonmin, latmin = transformer.transform(min(x), min(y))
    lonmax, latmax = transformer.transform(max(x), max(y))

    transformer = Transformer.from_crs("epsg:4326", proj, always_xy=True)

    print(x.shape, y.shape, usurf.shape)

    fsurf = RectBivariateSpline(x, y, np.transpose(usurf))

    df = pd.concat(
        [pd.read_csv(file) for file in files],
        ignore_index=True
    )
    mask = (
        (lonmin <= df["lon"])
        & (df["lon"] <= lonmax)
        & (latmin <= df["lat"])
        & (df["lat"] <= latmax)
        & df["elevation"].notnull()
        & df["date"].notnull()
        & df["elevation_date"].notnull()
    )
    df = df[mask]

    # Filter by date gap in second step for speed
    mask = (
        df["date"].str.slice(0, 4).astype(int) -
        df["elevation_date"].str.slice(0, 4).astype(int)
    ).abs().le(1)
    df = df[mask]

    # Compute thickness relative to prescribed surface
    xx, yy = transformer.transform(df["lon"], df["lat"])
    bedrock = df["elevation"] - df["thickness"]
    elevation_normalized = fsurf(xx, yy, grid=False)
    thickness_normalized = np.maximum(elevation_normalized - bedrock, 0)

    # Rasterize thickness
    thickness_gridded = pd.DataFrame({
        'col': np.floor(xx - np.min(x) / (x[1] - x[0])).astype(int),
        'row': np.floor(yy - np.min(y) / (y[1] - y[0])).astype(int),
        'thickness': thickness_normalized
    }).groupby(['row', 'col'])['thickness'].mean()
    thkobs = np.full((y.shape[0], x.shape[0]), np.nan)
    thkobs[tuple(zip(*thickness_gridded.index))] = thickness_gridded
    return thkobs
```

ho11laqe committed:
update layout

ho11laqe committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	examples/aletsch-basic/params.json

ho11laqe committed:
Merge remote-tracking branch 'origin/main'

ho11laqe committed:
- fixed ratio of the axis
- added sliders for camera position, camera position can be fixed so that it does not reset after changing the shown property
- layout_changes

ho11laqe committed:
- fixed ratio of the axis
- added sliders for camera position, camera position can be fixed so that it does not reset after changing the shown property
- layout_changes
